apps/ejabberd/src/gen_mod.erl

```diff
@@ -404,3 +414,17 @@ clear_opts(Module, Opts0) ->
         _ ->
             Opts
     end.
+
+-type db_type() :: odbc | mnesia | riak.
+
+-spec v_db(db_type() | internal) -> db_type().
+
+v_db(odbc) -> odbc;
+v_db(internal) -> mnesia;
+v_db(mnesia) -> mnesia;
+v_db(riak) -> riak.
+
+db_type(Host, Module) when is_atom(Module) ->
+    get_module_opt(Host, Module, db_type, mnesia);
```

Comment:
I'd like to be consistent with other our modules and use param `backend` instead of `db_type`.